cmake/FindESP.cmake

```diff
@@ -0,0 +1,256 @@
+#[=======================================================================[.rst:
+FindESP
+---------
```

Comment:
@tuckerbabcock is this something you wrote? The style of the commenting looks different which indicates it might come from elsewhere. If so, we need to include the source/licensing information.